app/code/core/Mage/Catalog/Model/Resource/Product/Action.php

```diff
@@ -81,4 +83,24 @@ public function updateAttributes($entityIds, $attrData, $storeId)
 
         return $this;
     }
+
+    /**
+     * Updated the "updated_at" field for all entity_ids passed to the method
+     * @param array $entityIds
+     * @return void
+     * @throws Zend_Db_Adapter_Exception
+     */
+    protected function _updateUpdatedAt(array $entityIds): void
+    {
+        $updatedAt = Varien_Date::now();
+        $catalogProductTable = $this->getTable('catalog/product');
+        $writeConnection = $this->getWriteConnection();
+
+        $entityIdsChunks = array_chunk($entityIds, 1000);
+        foreach ($entityIdsChunks as $entityIdsChunk) {
+            $writeConnection->update($catalogProductTable, [
+                'updated_at' => $updatedAt
+            ], $writeConnection->quoteInto('entity_id IN (?)', $entityIdsChunk));
```

Comment:
Style changes:
```suggestion
     * Update the "updated_at" field.
     *
     * @param array $entityIds
     * @return void
     * @throws Zend_Db_Adapter_Exception
     */
    protected function _updateUpdatedAt(array $entityIds): void
    {
        $updatedAt = Varien_Date::now();
        $catalogProductTable = $this->getTable('catalog/product');
        $adapter = $this->_getWriteAdapter();

        $chunks = array_chunk($entityIds, 1000);
        foreach ($chunks as $chunk) {
            $adapter->update(
                $catalogProductTable, 
                ['updated_at' => $updatedAt],
                $adapter->quoteInto('entity_id IN (?)', $chunk)
            );
```

Reply:
I kinda met you halfway 😅 what do you think about the new version?